File: +bsv/plotConnectivityWithSubregionAnalysis.m
function [projectionMatrix_array, projectionMatrixCoordinates_ARA, subregionResults, globalResults] = plotConnectivityWithSubregionAnalysis(experimentData, allenAtlasPath, allenAtlasPath_v2, outputRegion, numberOfChunks, numberOfPixels, plane, regionOnly, smoothing, colorLimits, color, normalizationInfo, inputRegions, regionGroups, experimentRegionInfo, normalizeByGroup, customSlices, sliceAveraging)
% plotConnectivityWithSubregionAnalysis - Plot connectivity and analyze CP subregions
%
% This function combines plotConnectivity with CP subregion analysis
%
% INPUTS: Same as plotConnectivity, plus:
%   allenAtlasPath_v2 - Path to v2 atlas files for subregion analysis
%
% OUTPUTS: Same as plotConnectivity, plus:
%   subregionResults - Per-slice subregion analysis
%   globalResults - Global (all slices) subregion analysis

% Handle optional parameters (same as plotConnectivity)
if nargin < 18 || isempty(sliceAveraging)
    sliceAveraging = 0;
end
if nargin < 17 || isempty(customSlices)
    customSlices = [];
end
if nargin < 16 || isempty(normalizeByGroup)
    normalizeByGroup = false;
end

fprintf('=== CONNECTIVITY ANALYSIS WITH CP SUBREGION ANALYSIS ===\n');

%% Step 1: Run normal connectivity plotting
fprintf('Step 1: Running connectivity analysis...\n');

[projectionMatrix_array, projectionMatrixCoordinates_ARA] = bsv.plotConnectivity(experimentData, allenAtlasPath, outputRegion, numberOfChunks, numberOfPixels, plane, regionOnly, smoothing, colorLimits, color, normalizationInfo, inputRegions, regionGroups, experimentRegionInfo, normalizeByGroup, customSlices, sliceAveraging);

%% Step 2: Analyze CP subregions
fprintf('\nStep 2: Analyzing CP subregions...\n');

if strcmpi(outputRegion, 'CP') || contains(upper(outputRegion), 'CP')
    [subregionResults, globalResults] = bsv.analyzeCPSubregions(projectionMatrix_array, projectionMatrixCoordinates_ARA, allenAtlasPath_v2, [], inputRegions, regionGroups);
    
    %% Step 3: Create summary table and save results
    fprintf('\nStep 3: Creating summary table...\n');
    
    % Create a summary table
    summaryTable = table();
    summaryTable.SubregionID = globalResults.subregion_ids;
    summaryTable.SubregionName = globalResults.subregion_names;
    summaryTable.SubregionAcronym = globalResults.subregion_acronyms;
    summaryTable.GlobalMeanIntensity = globalResults.mean_intensities;
    summaryTable.TotalVoxelCount = globalResults.total_voxel_counts;
    
    % Add per-slice statistics
    summaryTable.MeanAcrossSlices = mean(subregionResults.mean_intensities, 2, 'omitnan');
    summaryTable.StdAcrossSlices = std(subregionResults.mean_intensities, 0, 2, 'omitnan');
    summaryTable.SlicesWithData = sum(~isnan(subregionResults.mean_intensities), 2);
    
    % Remove subregions with no data
    hasData = ~isnan(summaryTable.GlobalMeanIntensity) | summaryTable.SlicesWithData > 0;
    summaryTable = summaryTable(hasData, :);
    
    % Sort by global mean intensity
    summaryTable = sortrows(summaryTable, 'GlobalMeanIntensity', 'descend', 'MissingPlacement', 'last');
    
    fprintf('\nCP Subregion Summary (top 10):\n');
    fprintf('%-30s %-15s %-15s %-10s\n', 'Subregion', 'Global Mean', 'Mean±Std', 'Slices');
    fprintf('%-30s %-15s %-15s %-10s\n', repmat('-', 1, 30), repmat('-', 1, 15), repmat('-', 1, 15), repmat('-', 1, 10));
    
    nToShow = min(10, height(summaryTable));
    for i = 1:nToShow
        if ~isnan(summaryTable.GlobalMeanIntensity(i))
            fprintf('%-30s %-15.4f %-15s %-10d\n', ...
                summaryTable.SubregionAcronym{i}, ...
                summaryTable.GlobalMeanIntensity(i), ...
                sprintf('%.4f±%.4f', summaryTable.MeanAcrossSlices(i), summaryTable.StdAcrossSlices(i)), ...
                summaryTable.SlicesWithData(i));
        end
    end
    
    % Optionally save the summary table
    % Uncomment the following lines if you want to save to CSV:
    % csvFilename = sprintf('CP_subregion_analysis_%s.csv', datestr(now, 'yyyymmdd_HHMMSS'));
    % writetable(summaryTable, csvFilename);
    % fprintf('\nSummary table saved to: %s\n', csvFilename);
    
else
    warning('Output region is not CP - skipping subregion analysis');
    subregionResults = [];
    globalResults = [];
end

fprintf('\n=== ANALYSIS COMPLETE ===\n');

end

File: bsv/plot_connectivity_with_subregion_analysis.py
from .plot_connectivity import plot_connectivity
from .analyze_cp_subregions import analyze_cp_subregions


def plot_connectivity_with_subregion_analysis(experiment_data, allen_atlas_path,
                                               allen_atlas_path_v2, output_region,
                                               number_of_chunks, number_of_pixels, plane,
                                               region_only, smoothing, color_limits, color,
                                               normalization_info='unknown', input_regions=None,
                                               region_groups=None, experiment_region_info=None,
                                               normalize_by_group=False, custom_slices=None,
                                               slice_averaging=0):
    """Plot projections and run CP subregion analysis in one step.

    Combines :func:`plot_connectivity` and :func:`analyze_cp_subregions`.

    Parameters
    ----------
    experiment_data : numpy.ndarray
        Projection density array from :func:`fetch_connectivity_data`.
    allen_atlas_path : str
        Path to the Allen CCF atlas directory.
    allen_atlas_path_v2 : str
        Path to the Allen v2 atlas directory.
    output_region : str
        Target region acronym (e.g. ``'CP'``).
    number_of_chunks : int
        Number of slices.
    number_of_pixels : int
        Pixel resolution per slice panel.
    plane : str
        ``'coronal'`` or ``'sagittal'``.
    region_only : bool
        Mask to the target region boundary.
    smoothing : float
        Gaussian smoothing sigma in pixels.
    color_limits : str or list
        Colour scale specification.
    color : list or None
        RGB colour(s).
    normalization_info : str, optional
        Normalization label.
    input_regions : list of str, optional
        Source region acronyms.
    region_groups : list of int, optional
        Group assignment per input region.
    experiment_region_info : dict, optional
        Per-experiment metadata.
    normalize_by_group : bool, optional
        Normalize each group independently.
    custom_slices : list of int, optional
        Specific slice indices.
    slice_averaging : int, optional
        Adjacent-slice averaging radius.

    Returns
    -------
    proj_array : numpy.ndarray
        Projection matrix.
    proj_coords : list
        Coordinate information.
    subregion_results : dict or None
        Subregion analysis results (None if region is not CP).
    global_results : dict or None
        Global subregion results (None if region is not CP).
    """
    print('=== CONNECTIVITY ANALYSIS WITH CP SUBREGION ANALYSIS ===')

    # Step 1: connectivity plotting
    print('Step 1: Running connectivity analysis...')
    proj_array, proj_coords = plot_connectivity(
        experiment_data, allen_atlas_path, output_region,
        number_of_chunks, number_of_pixels, plane,
        region_only, smoothing, color_limits, color,
        normalization_info, input_regions, region_groups,
        experiment_region_info, normalize_by_group,
        custom_slices, slice_averaging)

    # Step 2: subregion analysis
    subregion_results = None
    global_results = None

    region_str = output_region if isinstance(output_region, str) else output_region[0]
    if 'CP' in region_str.upper():
        print('\nStep 2: Analyzing CP subregions...')
        subregion_results, global_results = analyze_cp_subregions(
            proj_array, proj_coords, allen_atlas_path_v2,
            input_regions=input_regions, region_groups=region_groups)
    else:
        print(f'Warning: Output region is not CP - skipping subregion analysis')

    print('\n=== ANALYSIS COMPLETE ===')
    return proj_array, proj_coords, subregion_results, global_results
